rules/prefer-string-slice.js

```diff
@@ -23,10 +41,33 @@ const create = context => {
 				message: 'Prefer `String#slice()` over `String#substr()`.'
 			};
 
-			const canFix = argumentNodes.length === 0;
+			const firstArgument = argumentNodes[0] ? sourceCode.getText(argumentNodes[0]) : undefined;
+			const secondArgument = argumentNodes[1] ? sourceCode.getText(argumentNodes[1]) : undefined;
+
+			let slice;
+
+			if (argumentNodes.length === 0) {
+				slice = [];
+			} else if (argumentNodes.length === 1) {
+				slice = [firstArgument];
+			} else if (argumentNodes.length === 2) {
+				if (firstArgument === '0') {
```

Comment:
should be literal zero, we  don't want handle `"0"`, utils should have a function d this

Reply:
The text representation of an argument node can only ever be `=== '0'` if it's a literal zero, no?

Reply:
sorry, my comment is based on `const firstArgument = argumentNodes[0]`(I changed it in my refactor, but forgot it's not changed here), why we  need `sourceCode.getText`? Isn't `argumentNodes[0]` easier to use?

Reply:
As almost every branch used the value of `firstArgument` I decided to calculate them in  shared space to make the code more easy to read and more DRY.

As `firstArgument` then was already calculated I considered `firstArgument === '0'` to be preferable over `isLiteralNumber(argumentNodes[0]) && argumentNodes[0].value === 0` as it's shorter and simpler.

Reply:
thanks for explaining this, this make sense now, but I think it makes the logic little hard to understand. you want use those functions? 

https://github.com/sindresorhus/eslint-plugin-unicorn/blob/9e0db0b1385c72695d157f2d00bbc6a8ffdbbfc6/rules/prefer-includes.js#L3

https://github.com/sindresorhus/eslint-plugin-unicorn/blob/9e0db0b1385c72695d157f2d00bbc6a8ffdbbfc6/rules/prefer-includes.js#L10